packages/react-meteor-data/suspense/useTracker.ts

```diff
@@ -0,0 +1,305 @@
+import { Tracker } from 'meteor/tracker'
+import { EJSON } from 'meteor/ejson'
+import { DependencyList, useEffect, useMemo, useReducer, useRef } from 'react'
+import { Meteor } from 'meteor/meteor'
+
+function checkCursor(data: any): void {
+  let shouldWarn = false
+  if (Package.mongo && Package.mongo.Mongo && data && typeof data === 'object') {
+    if (data instanceof Package.mongo.Mongo.Cursor) {
+      shouldWarn = true
+    } else if (Object.getPrototypeOf(data) === Object.prototype) {
+      Object.keys(data).forEach((key) => {
+        if (data[key] instanceof Package.mongo.Mongo.Cursor) {
+          shouldWarn = true
+        }
+      })
+    }
+  }
+  if (shouldWarn) {
+    console.warn(
+      'Warning: your reactive function is returning a Mongo cursor. ' +
+      'This value will not be reactive. You probably want to call ' +
+      '`.fetch()` on the cursor before returning it.'
+    )
+  }
+}
+
+export const cache = new Map<string, Cache>()
+
+interface Cache {
+  deps: EJSON[]
+  promise: Promise<unknown>
+  result?: unknown
+  error?: unknown
+}
+
+// Used to create a forceUpdate from useReducer. Forces update by
+// incrementing a number whenever the dispatch method is invoked.
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+export type IReactiveFn<T> = (c?: Tracker.Computation) => Promise<T>
+
+export type ISkipUpdate<T> = <T>(prev: T, next: T) => boolean
+
+interface TrackerRefs {
+  computation?: Tracker.Computation
+  isMounted: boolean
+  trackerData: any
+}
+
+function resolveAsync(key: string, promise: Promise<unknown>, deps: DependencyList = []) {
+  const cached = cache.get(key) ?? false
```

Comment:
What's this `false` used for?

Reply:
in 
```js
 const cached = cache.get(key) ?? false
    useEffect(() => {
    return () => {
      setTimeout(() => {
        if (cached && isEqual(cached.deps, deps)) cache.delete(key)
        //  ˆˆˆˆˆ here is false in case we do not have it; 
      }, 0)
    }
  }, [cached, key, ...deps])
```
sure it could be left as undefined, but explicitly showing that it can be false sounds better. What do you think?

Reply:
I think it'd be better to leave it as `undefined` and either leave the condition as is, or add `!== undefined` there. `Entry | false` is a weird type.